src/Compatibility/Microsoft.DotNet.ApiCompatibility/ValidationSuppressions/SuppressionComparer.cs

```diff
@@ -0,0 +1,20 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.DotNet.ValidationSuppression
+{
+    /// <summary>
+    /// Defines methods to support the comparison of Suppression objects for equality.
+    /// </summary>
+    internal class SuppressionComparer : IEqualityComparer<Suppression>
+    {
+        /// <inheritdoc/>
+        public bool Equals(Suppression x, Suppression y) => x.Equals(y);
+
+        /// <inheritdoc/>
+        public int GetHashCode(Suppression obj) => HashCode.Combine(obj.DiagnosticId?.ToLowerInvariant(), obj.Target?.ToLowerInvariant(), obj.Left?.ToLowerInvariant(), obj.Right?.ToLowerInvariant());
```

Comment:
WHy not override `GetHashCode` on the suppression itself and just call that here?

Reply:
You are right, I'll do that and that way we can just use the GenericEqualityComparer